src/database/entities/server.rs

```diff
@@ -131,13 +130,81 @@ impl Server {
         Ok(())
     }
 
-    pub async fn publish_update(&self, data: Value) -> Result<()> {
-        ClientboundNotification::ServerUpdate {
-            id: self.id.clone(),
-            data,
-            clear: None,
+    pub async fn publish_update(&self, to_user: Option<&str>) -> Result<()> {
+        if let Some(user_id) = to_user {
+            let mut channels = Vec::new();
+
+            // Find the channels
+            let mut channel_cursor = get_collection("channels")
+                .find(
+                    doc! {
+                        "_id": {
+                            "$in": &self.channels
+                        }
+                    },
+                    None
+                )
+                .await
+                .map_err(|_| Error::DatabaseError {
+                    operation: "find",
+                    with: "channels",
+                })?;
+
+            // Only include channels the user can see
+            let user = User::from_id(user_id).await?;
+
+            while let Some(maybe_channel) = channel_cursor.next().await {
+                if let Ok(channel_document) = maybe_channel {
+                    let channel: Channel = from_document(channel_document)
+                        .map_err(|_| Error::DatabaseError {
+                            operation: "from_document",
+                            with: "channel",
+                        })?;
+
+                    let perms = PermissionCalculator::new(&user)
+                        .with_channel(&channel)
+                        .for_channel().await?;
+
+                    if perms.get_view() {
+                        channels.push(channel.id().into());
+                    }
+                }
+            }
+
+            ClientboundNotification::ServerUpdate {
+                id: self.id.clone(),
+                nonce: self.nonce.clone(),
+                owner: self.owner.clone(),
+                name: self.name.clone(),
+                description: self.description.clone(),
+                channels,
+                roles: self.roles.clone(),
+                default_permissions: self.default_permissions.clone(),
+                categories: self.categories.clone(),
+                system_messages: self.system_messages.clone(),
+                icon: self.icon.clone(),
+                banner: self.banner.clone(),
+                nsfw: self.nsfw.clone(),
+            }
+            .publish_as_user(user_id.into());
```

Comment:
`publish_as_user` is intended to take an event and distribute it to all server topics that the member is a part of as well as their own. This would actually send the server update to other irrelevant servers.

The event would also be from the perspective of the user despite going to other users.